linter warnings since pylint 3.2.0

Since [pylint version 3.2.0](https://github.com/pylint-dev/pylint/releases/tag/v3.2.0) was released, [our CI fails](https://github.com/labgrid-project/labgrid/actions/workflows/scheduled-unit-tests.yml) with these errors in the linting step (warnings emitted on current master 9e0562e1):

> ************* Module labgrid.util.yaml
> labgrid/util/yaml.py:87:18: E0606: Possibly using variable 'items' before assignment (possibly-used-before-assignment)

We should probably raise a TypeError in an else clause.

> ************* Module labgrid.util.dict
> labgrid/util/dict.py:39:33: E1133: Non-iterable value attr.fields(cls) is used in an iterating context (not-an-iterable)

Seems to be related to https://github.com/pylint-dev/pylint/issues/9626 ([attr's pyi](https://github.com/python-attrs/attrs/blob/main/src/attr/__init__.pyi)). Should be fixed in the next pylint release.

> ************* Module labgrid.util.agents.network_interface
> labgrid/util/agents/network_interface.py:98:23: E0606: Possibly using variable 'nm' before assignment (possibly-used-before-assignment)

Probably caused by..

https://github.com/labgrid-project/labgrid/blob/9e0562e1c8d6842c850c239d64896b88c3bf3bd1/labgrid/util/agents/network_interface.py#L291-L295

@jluebbe How should we fix this?

> ************* Module labgrid.remote.common
> labgrid/remote/common.py:178:24: E1135: Value 'fields' doesn't support membership test (unsupported-membership-test)

Might also be related to https://github.com/pylint-dev/pylint/issues/9626 ?

> ************* Module labgrid.remote.client
> labgrid/remote/client.py:1215:27: E0606: Possibly using variable 'data' before assignment (possibly-used-before-assignment)

I guess we should add an else clause raising an exception here as well?

> labgrid/remote/client.py:1327:29: E0606: Possibly using variable 'data' before assignment (possibly-used-before-assignment)

Same here.

> ************* Module labgrid.driver.mqtt
> labgrid/driver/mqtt.py:46:23: E0606: Possibly using variable 'status' before assignment (possibly-used-before-assignment)

Same here.

--- a/labgrid/driver/mqtt.py
+++ b/labgrid/driver/mqtt.py
@@ -43,6 +43,8 @@ class TasmotaPowerDriver(Driver, PowerProtocol):
             status = True
         elif msg.payload == b'OFF':
             status = False
+        else:
+            raise ValueError(f"Unknown status: {msg.payload}. Must be 'ON' or 'OFF'")
         self._status = status
 
     def _on_connect(self, client, userdata, flags, reason_code, properties):

--- a/labgrid/remote/client.py
+++ b/labgrid/remote/client.py
@@ -1212,6 +1212,9 @@ class ClientSession(ApplicationSession):
             data = drv.get_channel_info(channel)
         elif action == 'values':
             data = drv.get_channel_values(channel)
+        else:
+            raise ValueError(f"unknown action {action}")
+
         for k, v in sorted(data.items()):
             print(f"{k:<16s} {str(v):<10s}")
 
@@ -1323,6 +1326,9 @@ class ClientSession(ApplicationSession):
             data = "\n".join(lines) + "\n"
         elif self.args.format is ExportFormat.JSON:
             data = json.dumps(exported)
+        else:
+            raise NotImplementedError(f"unsupported format {self.args.format}")
+
         if self.args.filename == "-":
             sys.stdout.write(data)
         else:

--- a/labgrid/util/agents/network_interface.py
+++ b/labgrid/util/agents/network_interface.py
@@ -95,7 +95,7 @@ def connection_from_dict(data):
 class NMDev:
     def __init__(self, interface):
         self._interface = interface
-        self._nm_dev = nm.get_device_by_iface(interface)
+        self._nm_dev = nm.get_device_by_iface(interface) # pylint: disable=possibly-used-before-assignment
 
     def _delete_connection(self, con):
         future = Future()
@@ -114,7 +114,7 @@ class NMDev:
         )
 
     def get_settings(self):
-        lg_con = nm.get_connection_by_id(f"labgrid-{self._interface}")
+        lg_con = nm.get_connection_by_id(f"labgrid-{self._interface}") # pylint: disable=possibly-used-before-assignment
         if lg_con:
             return dict(lg_con.to_dbus(NM.ConnectionSerializationFlags.ALL))
 
@@ -125,7 +125,7 @@ class NMDev:
             return dict(con.to_dbus(NM.ConnectionSerializationFlags.ALL))
 
     def configure(self, data):
-        lg_con = nm.get_connection_by_id(f"labgrid-{self._interface}")
+        lg_con = nm.get_connection_by_id(f"labgrid-{self._interface}") # pylint: disable=possibly-used-before-assignment
         if lg_con:
             self._delete_connection(lg_con)
         data['connection'].update({
@@ -139,12 +139,12 @@ class NMDev:
         def cb(dev, res, error):
             assert error is None
             try:
-                res = nm.add_and_activate_connection_finish(res)
+                res = nm.add_and_activate_connection_finish(res) # pylint: disable=possibly-used-before-assignment
                 future.set(res)
             except Exception as e:
                 future.set(e)
 
-        nm.add_and_activate_connection_async(
+        nm.add_and_activate_connection_async( # pylint: disable=possibly-used-before-assignment
             con,
             self._nm_dev,
             None,  # specific_object
@@ -172,7 +172,7 @@ class NMDev:
             )
 
     def disable(self):
-        lg_con = nm.get_connection_by_id(f"labgrid-{self._interface}")
+        lg_con = nm.get_connection_by_id(f"labgrid-{self._interface}") # pylint: disable=possibly-used-before-assignment
         if lg_con:
             self._delete_connection(lg_con)
 

--- a/labgrid/util/yaml.py
+++ b/labgrid/util/yaml.py
@@ -84,6 +84,9 @@ def resolve_templates(data, mapping):
         items = enumerate(data)
     elif isinstance(data, dict):
         items = data.items()
+    else:
+        raise TypeError(f"Expected list or dict, got {type(data)}")
+
     for k, val in items:
         if isinstance(val, Template):
             try:

